app/controllers/assessment/autograde.rb

```diff
@@ -29,30 +24,31 @@ def autogradeAfterHandin(submission)
         flash[:error] += " Please contact your instructor."
       end
     elsif job == -1 then
-      flash[:error] = "There was an error (#{job}) submitting your autograding job. " +
-        "Please contact your instructor."
+      link = "<a href=\"#{url_for(:controller=>'jobs')}\">Jobs</a>"
+      flash[:error] = "There was an error submitting your autograding job. " +
+        "Check the #{link} page for more info."
     else
       link = "<a href=\"#{url_for(:controller=>"jobs", :action=>"getjob", :id=>job)}\">Job ID = #{job}</a>"
-      flash[:success] = ("Submitted file #{@submission.filename} (#{link}) for autograding." +
+      flash[:success] = ("Submitted file #{submissions[0].filename} (#{link}) for autograding." +
         " Refresh the page to see the results.").html_safe
     end
+    return job
   end
 
-# 
+  ##
   # sendJob - this scary-looking function initiates an autograding
   # job request on the backend. It builds a job structure that
   # contains various info about the job, send submits it to the
   # Tango server via an REST API.
   #
-  # Required global input variables:
-  # @course, @assessment, @submission
+  # submissions must have at least one element
   #
-  def sendJob
-    extend_config_module()
-    assessmentDir = File.join(AUTOCONFIG_COURSE_DIR,@course.name,@assessment.name)
+  def sendJob(course, assessment, submissions)
```

Comment:
So for groups, we only want to submit one job to be autograded, because, at least with compilers, jobs can take a really long time and there's no need to grade the same submission twice.  But we still need to pick a submission to be the canonical submission that actually gets autograded, so we just go with the first one, since it's guaranteed to be there.